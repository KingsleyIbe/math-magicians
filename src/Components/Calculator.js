import React, { useState } from 'react';
import '../styles/App.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: null, next: 0, operation: null });
  const handleBtnClick = (e) => {
    setState(calculate(state, e.target.textContent));
  };

  return (
    <div id="container">
      <div className="calc-h2">
        <h2>Lets do some math!</h2>
      </div>
      <div id="calculator">
        <div id="result">
          <div id="history">
            <p id="history-value" />
          </div>
          <div id="output">
            { state.total }
            { state.operation }
            { state.next }
          </div>
        </div>
        <div id="keyboard">
          <button type="button" className="operator" id="clear" onClick={handleBtnClick}>C</button>
          <button type="button" className="operator" id="backspace" onClick={handleBtnClick}>AC</button>
          <button type="button" className="operator" id="backspace" onClick={handleBtnClick}>+/-</button>
          <button type="button" className="operator" id="+" onClick={handleBtnClick}>&#247;</button>
          <button type="button" className="operator" id="7" onClick={handleBtnClick}>7</button>
          <button type="button" className="number" id="8" onClick={handleBtnClick}>8</button>
          <button type="button" className="number" id="9" onClick={handleBtnClick}>9</button>
          <button type="button" className="number" id="*" onClick={handleBtnClick}>x</button>
          <button type="button" className="operator" id="4" onClick={handleBtnClick}>4</button>
          <button type="button" className="number" id="5" onClick={handleBtnClick}>5</button>
          <button type="button" className="number" id="6" onClick={handleBtnClick}>6</button>
          <button type="button" className="number" id="7" onClick={handleBtnClick}>-</button>
          <button type="button" className="operator" id="1" onClick={handleBtnClick}>1</button>
          <button type="button" className="number" id="2" onClick={handleBtnClick}>2</button>
          <button type="button" className="number" id="3" onClick={handleBtnClick}>3</button>
          <button type="button" className="number" id="4" onClick={handleBtnClick}>+</button>
          <button type="button" className="operator" id="+" onClick={handleBtnClick}>%</button>
          <button type="button" className="empty" id="0" onClick={handleBtnClick}>0</button>
          <button type="button" className="number" id="." onClick={handleBtnClick}>.</button>
          <button type="button" className="operator" id="=" onClick={handleBtnClick}>=</button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
