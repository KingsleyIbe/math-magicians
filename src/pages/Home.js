import React from 'react';

const Home = () => (
  <section>
    <div>
      <h1 className="home">Welcome to our page</h1>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras gravida,
        risus at dapibus aliquet, elit quam scelerisque tortor, nec accumsan eros
        nulla interdum justo. Pellentesque dignissim, sapien et congue rutrum,
        lorem tortor dapibus turpis, sit amet vestibulum eros mi et odio.
      </p>
    </div>
  </section>
);

export default Home;
