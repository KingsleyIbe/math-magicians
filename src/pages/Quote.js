import { useEffect, useState } from 'react';

const Quote = () => {
  const defaultQuote = 'Without mathematics, thereâ€™s nothing you can do. Everything around you is mathematics. Everything around you is numbers. - Shakuntala Devi';
  const [quote, setQuote] = useState(`${defaultQuote}`);

  useEffect(() => {
    async function getQuotes() {
      const res = await fetch('https://random-math-quote-api.herokuapp.com/');
      const data = await res.json();
      const quotes = `${data.quote} - ${data.author}`;
      const timer = setTimeout(() => setQuote(quotes), 5000);
      return () => {
        clearTimeout(timer);
      };
    }
    getQuotes();
  }, [quote]);

  return (
    <section className="quote-section">
      <div className="home">
        <div>
          <h2>Random Mathematics Quotes</h2>
        </div>
        <p>{quote}</p>
      </div>
    </section>
  );
};

export default Quote;
