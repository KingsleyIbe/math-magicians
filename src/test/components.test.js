import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from 'src/components/Calculator';
import Home from '../pages/Home';
import Quote from '../pages/Quote';

describe('Test UI of the webpage', () => {
  it('Home displays on the page', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Calculator displays on the page', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Quote displays on the page', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
