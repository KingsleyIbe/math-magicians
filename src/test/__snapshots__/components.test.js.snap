// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UI of the webpage Calculator displays on the page 1`] = `
<div
  id="container"
>
  <div
    className="calc-h2"
  >
    <h2>
      Let's do some math!
    </h2>
  </div>
  <div
    id="calculator"
  >
    <div
      id="result"
    >
      <div
        id="history"
      >
        <p
          id="history-value"
        />
      </div>
      <div
        id="output"
      >
        0
      </div>
    </div>
    <div
      id="keyboard"
    >
      <button
        className="operator"
        id="clear"
        onClick={[Function]}
        type="button"
      >
        C
      </button>
      <button
        className="operator"
        id="backspace"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="operator"
        id="backspace"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="operator"
        id="+"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="operator"
        id="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="number"
        id="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="number"
        id="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="number"
        id="*"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="operator"
        id="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="number"
        id="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="number"
        id="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="number"
        id="7"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="operator"
        id="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="number"
        id="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="number"
        id="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="number"
        id="4"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="operator"
        id="+"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="empty"
        id="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="number"
        id="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operator"
        id="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Test UI of the webpage Home displays on the page 1`] = `
<section
  className="home-section"
>
  <div>
    <h2
      className="home"
    >
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras gravida, risus at dapibus aliquet, elit quam scelerisque tortor, nec accumsan eros nulla interdum justo. Pellentesque dignissim, sapien et congue rutrum, lorem tortor dapibus turpis, sit amet vestibulum eros mi et odio. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras gravida, risus at dapibus aliquet, elit quam scelerisque tortor, nec accumsan eros nulla interdum justo. Pellentesque dignissim, sapien et congue rutrum, lorem tortor dapibus turpis, sit amet vestibulum eros mi et odio.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras gravida, risus at dapibus aliquet, elit quam scelerisque tortor, nec accumsan eros nulla interdum justo. Pellentesque dignissim, sapien et congue rutrum, lorem tortor dapibus turpis, sit amet vestibulum eros mi et odio. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras gravida, risus at dapibus aliquet, elit quam scelerisque tortor, nec accumsan eros nulla interdum justo. Pellentesque dignissim, sapien et congue rutrum, lorem tortor dapibus turpis, sit amet vestibulum eros mi et odio.
    </p>
  </div>
</section>
`;

exports[`Test UI of the webpage Quote displays on the page 1`] = `
<section
  className="quote-section"
>
  <div
    className="home"
  >
    <div>
      <h2>
        Random Mathematics Quotes
      </h2>
    </div>
    <p>
      Without mathematics, there’s nothing you can do. Everything around you is mathematics. Everything around you is numbers. - Shakuntala Devi
    </p>
  </div>
</section>
`;
